src/server/managers/AppExternalComponentManager.ts

```diff
@@ -0,0 +1,74 @@
+import { IExternalComponent } from '../../definition/externalComponent';
+
+/**
+ * The external component manager for the Apps.
+ *
+ * An App will register external component during their `initialize` method.
+ * Then once an App's `onEnable` is called and it returns true,
+ * only then will that App's external component be enabled.
+ */
+export class AppExternalComponentManager {
+    /**
+     * The map that maintain the list of all registered components.
+     */
+    private providedComponents: Map<string, IExternalComponent>;
+    /**
+     * Contains the apps and the external components they have touhed.
+     * The key is appId and the value is external component.
+     * Doesn't matter whether the app provided, modified, disabled,
+     * or enabled. As long as an app touched an external component, then
+     * it is listed here.
+     */
+    private appTouchedComponents: Map<string, IExternalComponent>;
+    constructor() {
```

Comment:
```suggestion

    constructor() {
```

Reply:
resolved in https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/149/commits/ba0118e1832aceaad8285dd005ea4a05bdba4135